ipfx/epochs.py

```diff
@@ -86,53 +86,59 @@ def get_sweep_epoch(response):
     return 0, len(response)-1
 
 
-def get_stim_epoch_A(i,test_pulse=True):
-
+def get_stim_epoch(i, test_pulse=True):
     """
-    Identify start index, and end index of a general stimulus.
+    Determine the start index, and end index of a general stimulus.
+
+    Parameters
+    ----------
+    i   : numpy array
+        current
+
+    test_pulse: bool
+        True if test pulse is assumed
+
+    Returns
+    -------
+    start,end: int tuple
     """
 
     di = np.diff(i)
     di_idx = np.flatnonzero(di)   # != 0
 
-    start_idx_idx = 2 if test_pulse else 0     # skip the first up/down (test pulse) if present
+    if test_pulse:
+        di_idx = di_idx[2:]     # drop the first up/down (test pulse) if present
 
-    if len(di_idx[start_idx_idx:]) == 0:    # if no stimulus is found
+    if len(di_idx) == 0:    # if no stimulus is found
         return None, None
 
-    start_idx = di_idx[start_idx_idx] + 1   # shift by one to compensate for diff()
+    start_idx = di_idx[0] + 1   # shift by one to compensate for diff()
     end_idx = di_idx[-1]
 
     return start_idx, end_idx
 
 
-def get_stim_epoch_B(i):
-    """
-    Identify start index, and end index of a general stimulus. Assume there is test pulse.
-    """
+def get_test_epoch(i):
 
     di = np.diff(i)
     di_idx = np.flatnonzero(di)  # != 0)
```

Comment:
drop the commented out code fragment as you are touching this function

Reply:
Not sure what are you asking here...

Reply:
@sgratiy `  # != 0)` is commented out code and I think can be removed.

Reply:
Done